### PR TITLE
Feature/dictionary index

### PR DESCRIPTION
**Pull request by cardoni**
> **WARNING**
> Merge in #1 before this PR
> **/WARNING**

---
## Adds:
- Enables actual functionality of the `/index` route ;)

### CODE REVIEW

**Comment by vyder** [comment]
I don't think it's wise to add a `/` in the branch name, since that's what git uses to delimit the branch path like: `refs/remotes/origin/master` etc. Took me a min to find this branch.

File: app/app.js [hunk 1]
```diff
@@ -15,11 +15,12 @@ var dictionary = new Dictionary( {
 } );
 
 // Routes
-app.get( '/', require( './route-index' ) );
-app.get( '/hello', require( './route-hello' ) );
```
Comment (vyder) [inline app/app.js:19]: Do you want to hose the hello route file and update the README too?

File: app/app.js [hunk 2]
```diff
@@ -15,11 +15,12 @@ var dictionary = new Dictionary( {
...
+app.get( '/', require( './route-index' )( dictionary ) );
+app.get( '/index', require( './route-index' )( dictionary ) );
 app.get( '/lookup', require( './route-lookup' )( dictionary ) );
 app.get( '/add', require( './route-add' )( dictionary ) );
 app.get( '/remove', require( './route-remove' )( dictionary ) );
```
Comment (vyder) [inline app/app.js:22]: Passing in the dictionary to the route like this works, but it's too tightly coupled IMO. The route shouldn't need to know anything about dictionaries or implementations or anything. We can run that refactor as a separate issue.

File: app/app.js (hunk 1 above)
Comment (cardoni) [inline app/app.js:19]: Done and done — in PR #5.

File: app/app.js (hunk 2 above)
Comment (cardoni) [inline app/app.js:22]: I agree.  In the meantime, I vote that we merge this in as-is and revisit in the future (via your issue #4).

**Comment by cardoni** [comment]
> I don't think it's wise to add a / in the branch name, since that's what git uses to delimit the branch path like: refs/remotes/origin/master etc. Took me a min to find this branch.

@vyder: I added some thoughts on this to PR #5:
**tl;dr** - I'm in favor of branch-names with slashes in them, as I don't currently see what the issue is; but I could be wrong.
